packages/core/src/indexing/index-api.ts

```diff
@@ -4,14 +4,47 @@ import type {
   Page,
   Pagination,
   StreamState,
+  DiagnosticsLogger,
 } from '@ceramicnetwork/common'
+import { SyncOptions } from '@ceramicnetwork/common'
 import type { DatabaseIndexApi } from './database-index-api.js'
-import { NotImplementedError } from './not-implemented-error.js'
+import type { Repository } from '../state-management/repository.js'
 
+/**
+ * API to query an index.
+ */
 export class IndexApi implements IndexClientApi {
-  constructor(private readonly databaseIndexApi: DatabaseIndexApi | undefined) {}
+  constructor(
+    private readonly databaseIndexApi: DatabaseIndexApi | undefined,
+    private readonly repository: Repository,
+    private readonly logger: DiagnosticsLogger
+  ) {}
 
-  queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
-    throw new NotImplementedError('IndexApi::queryIndex')
+  /**
+   * Query the index. Ask an indexing database for a list of StreamIDs,
+   * and convert them to corresponding StreamState instances via `Repository::streamState`.
+   *
+   * We assume that a state store always contains StreamState for an indexed stream.
+   */
+  async queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
+    if (this.databaseIndexApi) {
+      const page = await this.databaseIndexApi.page(query)
+      const streamStates = await Promise.all(
+        page.entries.map((streamId) => this.repository.streamState(streamId))
```

Comment:
`Repository.streamState()` doesn't use the loading queue or add the state to the cache.  Should we be using `Repository.get()` instead?  https://github.com/ceramicnetwork/js-ceramic/blob/e8947c9ae2d7b050e34fdbd69cd33e0c80173050/packages/core/src/state-management/repository.ts#L268

Reply:
I more buy the argument that we should skip the loading queue since we know we won't be generating pubsub queries so the need to rate limit is less.  I'm a little less convinced that we should skip caching streams that are loaded via queries.  I do see the concern though that one large query could blow away your entire cache in one fell swoop.  So I'm okay with going with this for now and we can revisit in the future if this causes issues.